packages/client/hmi-client/src/components/data-explorer/search-bar.vue

```diff
@@ -3,126 +3,112 @@
 		<div class="search">
 			<span class="p-input-icon-left p-input-icon-right">
 				<i class="pi pi-search" />
-				<InputText
-					type="text"
-					placeholder="Search"
+				<AutoComplete
+					ref="searchBarRef"
+					:active="searchBarRef?.overlayVisible"
 					v-model="query"
-					@keyup.enter="execSearch"
-					@keyup.space="handleSearchEvent"
-					class="input-text"
-					@input="handleSearchEvent"
-					ref="inputElement"
-				/>
-				<Menu ref="autocompleteMenu" :model="autocompleteMenuItems" :popup="true"> </Menu>
-				<i
-					class="pi pi-times clear-search"
-					:class="{ hidden: isClearQueryButtonHidden }"
-					style="font-size: 1rem"
-					@click="clearQuery"
-				/>
+					:suggestions="autocompleteMenuItems"
+					placeholder="Search"
+					:autoOptionFocus="false"
+					:minLength="3"
+					scrollHeight="400px"
+					loadingIcon="null"
+					@complete="handleSearchEvent"
+					@keyup.enter="initiateSearch"
+					@item-select="initiateSearch"
+				>
+					<template #option="prop">
+						<span class="auto-complete-term">
+							<i class="pi pi-search" />
+							<span>{{ prop.option }}</span>
+							<i class="pi pi-arrow-right"
+						/></span>
+					</template>
+				</AutoComplete>
+				<i class="pi pi-times clear-search" :class="{ hidden: !query }" @click="clearQuery" />
 			</span>
 			<!-- <i class="pi pi-history" title="Search history" /> -->
-			<!-- <i class="pi pi-image" title="Search by Example" @click="toggleSearchByExample" /> -->
+			<!-- <i class="pi pi-image" title="Search by Example" @click="toggleSearchByExample" /> 
+					 @keyup.space="handleSearchEvent" @change="handleSearchEvent"-->
 		</div>
 	</section>
 </template>
 
 <script setup lang="ts">
-import { computed, onMounted, ref } from 'vue';
-import { useRoute } from 'vue-router';
-import InputText from 'primevue/inputtext';
+// import { isEmpty } from 'lodash';
+import { useRouter, useRoute } from 'vue-router';
+import { getRelatedTerms, getAutocomplete } from '@/services/data';
+import { RouteName } from '@/router/routes';
+import { onMounted, ref } from 'vue';
 import * as EventService from '@/services/event';
 import useResourcesStore from '@/stores/resources';
-import { getRelatedWords, getAutocomplete } from '@/services/data';
-import Menu from 'primevue/menu';
+import AutoComplete from 'primevue/autocomplete';
 import { EventType } from '@/types/EventType';
 
-const emit = defineEmits(['query-changed', 'toggle-search-by-example']);
-
-const route = useRoute();
-const resources = useResourcesStore();
-
 const props = defineProps<{
 	suggestions: boolean;
 }>();
 
-const query = ref('');
-const autocompleteMenu = ref();
-const autocompleteMenuItems = ref([{}]);
-const inputElement = ref<HTMLInputElement | null>(null);
+const emit = defineEmits(['update-related-terms', 'toggle-search-by-example']);
```

Comment:
This is where we might disagree. I prefer `query-changed` instead of `update-related-terms`.
- With `update-related-terms`, the component is aware of what function its parent component will trigger with this event.
- With `query-changed` (or `query-updated`, I don't care), the event becomes pure information, and it's up to the parent component to decide what to do with this information. For example, in a future PR, when the query is null, we will send the user back to his previous page (home or project), and that's up to the `NavBar` to decide, not the `search-bar`.

Reply:
I agree!